### PR TITLE
Fix reference tests

### PR DESCRIPTION
**Pull request by hjoliver**
~~UID~~ #4578 follow-up - "reference tests" cannot fail!!  :man_facepalming: 

This fixes that, and several functional test failures that were hidden by this bug.

This is a small change with no associated Issue.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Already covered by existing tests.
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: tests/functional/cylc-cat-log/03-bad-workflow.t
Comment (hjoliver) [inline tests/functional/cylc-cat-log/03-bad-workflow.t:23]: (Unrelated: this test was flaky - mktemp can generate strings starting with digits, resulting in "invalid workflow name" errors instead of the intended ones).

**Comment by MetRonnie** [comment]
`tests/k/cylc-poll/16-execution-time-limit.t` also failing for me locally on this branch

File: tests/functional/broadcast/00-simple/flow.cylc
Comment (oliver-sanders) [inline tests/functional/broadcast/00-simple/flow.cylc:97]: Remove the `--debug` and the problem goes away.

**Comment by oliver-sanders** [comment]
> UID follow-up

Is this a UID issue?

> tests/k/cylc-poll/16-execution-time-limit.t also failing for me locally on this branch

The test workflow isn't being run in `--debug` mode so the triggering isn't being logged.

> Let's not compare stderr here; it can be contaminated by debug output from parsec (repeated config items)

There's another option, we could log the trigger lines at info level when in `--reftest` mode.

**Review by oliver-sanders — APPROVED** [review]
Pre-emptive approval with https://github.com/hjoliver/cylc-flow/pull/20 or similar.

**Comment by hjoliver** [comment]
>  >  UID follow-up
> 
> Is this a UID issue?

Yes, the `_load_reflog` function was looking for triggering info matching the pre-UID task ID format. So the result was `ref-log == test-log` is `True` (via `[] == []`) for ALL reference tests.

File: tests/functional/broadcast/00-simple/flow.cylc
Comment (hjoliver) [reply tests/functional/broadcast/00-simple/flow.cylc:97]: > Remove the --debug and the problem goes away.

And so does the triggering info, so the reference test fails.

I figured there should not be any problem with the stderr file if the stdout is 100% correct. (However, I didn't check the details).

**Comment by hjoliver** [comment]
> > There's another option, we could log the trigger lines at info level when in --reftest mode.
> 
> hjoliver#20

Agreed that's a good option. Not sure why we didn't do that in the first place.

File: tests/functional/broadcast/00-simple/flow.cylc
Comment (hjoliver) [reply tests/functional/broadcast/00-simple/flow.cylc]: (merged your log-level change, so this is moot point now anyway!)

**Comment by hjoliver** [comment]
Merged the side PR. One flaky test failed, passes locally. No need for a second review on this.

**Pull request merged**
